pkg/perf/perf_suite_test.go

```diff
@@ -0,0 +1,203 @@
+	singleImageParallelPushTime int
+	singleImageParallelPullTime int
+	times                       int
+	parallelImagesNumber        int
+	numCPU                      int
+)
+
+// nolint: gochecknoinits
+func init() {
+	// Define the number of CPUs
+	// flag.IntVar(numCPU, "cpu", runtime.NumCPU(), "number of CPUs")
+	// numCPU = runtime.NumCPU()
+	// Define flags
+	flag.StringVar(&serverConfigPath, "server.config", "",
+		"path to the server config file")
+	flag.StringVar(&measurementConfigPath, "measurement.config", "",
+		"path to the measurement config file")
+}
+
+func ReadConfig() {
+	serverConfig = LoadServerConfig(serverConfigPath)
+	username = serverConfig.Username
+	password = serverConfig.Password
+	address = serverConfig.Address
+	repo = serverConfig.Repo
+
+	measurementConfig = LoadMeasurementConfig(measurementConfigPath)
+	singleImagePushTime = measurementConfig.SingleImagePushTime
+	singleImagePullTime = measurementConfig.SingleImagePullTime
+	parallelPushTime = measurementConfig.ParallelPushTime
+	singleImageParallelPushTime = measurementConfig.SingleImageParallelPushTime
+	singleImageParallelPullTime = measurementConfig.SingleImageParallelPullTime
+	times = measurementConfig.Times
+	parallelImagesNumber = measurementConfig.ParallelImagesNumber
+}
+
+func SetupFunc() {
+	if err := CreateImages(parallelImagesNumber); err != nil {
+		log.Fatalln("Error creating images!", err)
+	}
+
+	if err := os.MkdirAll("pulled_images", 0777); err != nil {
+		log.Fatalln("Error creating pulled_images dir!", err)
+	}
+}
+
+func CleanUpFunc() {
+	repoURL := fmt.Sprintf("https://%s", path.Join(address, repo))
+
+	_, err := resty.R().SetBasicAuth(username, password).Delete(repoURL)
+	if err != nil {
+		log.Fatalln("Error removing repository!", err)
+	}
+
+	if err = os.RemoveAll("pulled_images"); err != nil {
+		log.Fatalln("Error removing pulled_images dir!", err)
+	}
+
+	if err := DeleteImages(parallelImagesNumber); err != nil {
+		log.Fatalln("Error deleting images!", err)
+	}
+}
+
+func TestPerf(t *testing.T) {
+	_, errServer := os.Stat(serverConfigPath)
+	_, errMeasure := os.Stat(measurementConfigPath)
+
+	if errors.Is(errServer, os.ErrNotExist) && errors.Is(errMeasure, os.ErrNotExist) {
+		t.Skip("Skip running! Check server_config.json and measurement_config.json files!")
+	}
+
+	ReadConfig()
+	SetupFunc()
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Performance Suite")
+
+	// clean up time...
+	CleanUpFunc()
+}
+
+var _ = Describe("Check Zot Performance", func() {
+	Measure("Performance measuring - push", func(b Benchmarker) {
+		imageName := "zot-tests-dummy-push"
+
+		runtime := b.Time("runtime", func() {
+			Expect(PushImage(path.Join("images", imageName), username, password, address, repo, imageName)).To(BeNil())
+		})
+		Expect(runtime.Seconds()).To(BeNumerically("<", singleImagePushTime),
+			"Push oprations shouldn't take too long!")
+	}, times)
+
+	Measure("Performance measuring - parallel push", func(b Benchmarker) {
```

Comment:
No need to keep saying "Performance measuring" everywhere.
Just say "Parallel Monolithic Push" ... etc